prow/github/client.go

```diff
@@ -186,40 +186,42 @@ func (c *Client) Throttle(hourlyTokens, burst int) {
 }
 
 // NewClient creates a new fully operational GitHub client.
-// 'token' is the GitHub access token to use.
+// 'getToken' is a generator for the GitHub access token to use.
 // 'bases' is a variadic slice of endpoints to use in order of preference.
 //   An endpoint is used when all preceding endpoints have returned a conn err.
 //   This should be used when using the ghproxy GitHub proxy cache to allow
 //   this client to bypass the cache if it is temporarily unavailable.
-func NewClient(token string, bases ...string) *Client {
+func NewClient(getToken func() []byte, bases ...string) *Client {
 	return &Client{
 		logger: logrus.WithField("client", "github"),
 		time:   &standardTime{},
-		gqlc:   githubql.NewClient(oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token}))),
-		client: &http.Client{},
-		token:  token,
-		bases:  bases,
-		dry:    false,
+		gqlc: githubql.NewClient(oauth2.NewClient(context.Background(),
+			oauth2.StaticTokenSource(&oauth2.Token{AccessToken: string(getToken())}))),
```

Comment:
This calls `getToken` only once during client creation, but we need it to be called every time a githubql client call is made in order to pick up any changes to the token.

Reply:
another issue is already opened for it. #8702